src/ecdsa.rs

```diff
@@ -66,17 +72,40 @@ impl Signature {
     }
 
     /// Verify an ECDSA signature
-    pub fn verify(self, ctx: &Context, hash: &[u8], pub_key: &PubKey) -> bool {
+    pub fn verify(&self, ctx: &Context, hash: &[u8], pub_key: &PubKey) -> bool {
```

Comment:
Good catch!  I totally missed this one.

Reply:
Me too XD I only realized because when I tried to use the signature after calling verify in a test...bad things happened :P